tests/measurements/test_state.py

```diff
@@ -534,6 +558,19 @@ def func():
 
         assert np.allclose(state_expected, state_val)
 
+    def test_return_type_is_complex(self):
+        """Test that state always returns a complex value."""
+        dev = qml.devices.DefaultQubit()
```

Comment:
Should we unit test this instead of integration testing it?

Reply:
I do both - see the changes above. I think this integ test doesn't hurt